Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -131,13 +124,29 @@ private func expandFreestandingExpr(
     return nil
   }
 
-  let indentedSource =
-    expanded
-    .indented(by: node.indentationOfFirstLine)
-    .wrappingInTrivia(from: node)
+  let indentedSource = adjustIndentationOfFreestandingMacro(expandedCode: expanded, node: node)
+
   return "\(raw: indentedSource)"
 }
 
+/// Adds the appropriate indentation on expanded code even if it's multi line.
+/// Makes sure original macro expression's trivia is maintained by adding it to expanded code.
+private func adjustIndentationOfFreestandingMacro(expandedCode: String, node: some FreestandingMacroExpansionSyntax) -> String {
+  let indentationOfFirstLine = node.indentationOfFirstLine
+
+  var indentedSource =
+    expandedCode
+    .indented(by: indentationOfFirstLine)
+
+  if indentedSource.count >= indentationOfFirstLine.sourceLength.utf8Length {
+    indentedSource.removeFirst(indentationOfFirstLine.sourceLength.utf8Length)
+  }
```

Comment:
Am I missing something or does this check not make any sense? Previously you were checking `!node.leadingTrivia.contains(where: \.isNewline)`, which made sense to me but AFAICT this check is always satisfied because `indentedSource` has at least as many characters as the indentation of its first line.

Reply:
since we have
```
func indented(by indentation: Trivia) -> String {
    if isEmpty || indentation.isEmpty {
      return self
    }
    // so on.. 
```
when `expandedCode` is empty string, it returns empty string and calling `removeFirst(n)` causes a crash. [Test `testErrorExpansion()` crashes because of this]

---

about previous condition, I am using different approach altogether now 😅 , so this condition is not replacement of previous one.

now we are indenting the entire expanded code first and then un-indenting the first line (with removeFirst())
then we are wrapping the expanded code in macro's trivia.

Notice that, now `wrappingInTrivia(from:)` does not remove indentation on last line. It simply adds entire trivia of macro to expanded code as it is.

So, previously we were removing indentation on last line from macro's trivia and now we are removing indentation from first line on expanded indented code

---

Here are couple of examples, (assume numbers are spaces)
|Step|Single line macro| Multi line macro|
|--|--|--|
|expanded code | `"expanded_code"` | `{\n"expanded_code"\n}`
|indent the entire code | `12"expanded_code"` | `12{\n1234"expanded_code"\n12}`
|remove first line indentation| `"expanded_code"` | `{\n1234"expanded_code"\n12}`
|adding exact same trivia as macro node at new line| `12"expanded_code"` | `12{\n1234"expanded_code"\n12}`
|or adding exact same trivia as macro in middle of the line | `"expanded_code"` | `{\n1234"expanded_code"\n12}`

Reply:
Oh, I see. I forgot that we would essentially add the indentation of the first line again from `wrappingInTrivia`. Could you add a short version of that explanation as a comment for anyone else who reads this in the future?